packages/perseus/src/widgets/interactive-graphs/reducer/interactive-graph-reducer.ts

```diff
@@ -545,22 +550,90 @@ function boundAndSnapToAngle(
 
     const knownSide = magnitude(
         // @ts-expect-error - TS2345 - Argument of type 'number[]' is not assignable to parameter of type 'readonly Coord[]'.
-        vector(coords[rel(-1)], coords[rel(1)]),
+        vector(coordsCopy[rel(-1)], coordsCopy[rel(1)]),
     );
 
     const onLeft =
-        sign(ccw(coords[rel(-1)], coords[rel(1)], coords[index])) === 1;
+        sign(
+            ccw(coordsCopy[rel(-1)], coordsCopy[rel(1)], coordsCopy[index]),
+        ) === 1;
 
     // Solve for side by using the law of sines
     const side =
         (Math.sin((innerAngles[1] * Math.PI) / 180) /
             Math.sin((innerAngles[2] * Math.PI) / 180)) *
         knownSide;
 
-    const outerAngle = GraphUtils.findAngle(coords[rel(1)], coords[rel(-1)]);
+    const outerAngle = GraphUtils.findAngle(
+        coordsCopy[rel(1)],
+        coordsCopy[rel(-1)],
+    );
 
     const offset = polar(side, outerAngle + (onLeft ? 1 : -1) * innerAngles[0]);
-    return kvector.add(coords[rel(-1)], offset);
+    return kvector.add(coordsCopy[rel(-1)], offset) as vec.Vector2;
+}
+
+function boundAndSnapToSides(
+    destinationPoint: vec.Vector2,
+    {range, coords}: {range: [Interval, Interval]; coords: Coord[]},
+    index: number,
+) {
+    const startingPoint = coords[index];
+    const coordsCopy = [...coords];
+
+    // Takes the destination point and makes sure it is within the bounds of the graph
+    // SnapStep is [0, 0] because we don't want to snap to the grid
+    coordsCopy[index] = bound({
+        snapStep: [0, 0],
+        range,
+        point: destinationPoint,
+    });
+
+    // Gets the relative index of a point
+    const rel = (j): number => {
+        return (index + j + coordsCopy.length) % coordsCopy.length;
+    };
+    const sides = _.map(
+        [
+            [coordsCopy[rel(-1)], coordsCopy[index]],
+            [coordsCopy[index], coordsCopy[rel(1)]],
+            [coordsCopy[rel(-1)], coordsCopy[rel(1)]],
+        ],
+        function (coordsCopy) {
+            // @ts-expect-error - TS2345 - Argument of type 'number[]' is not assignable to parameter of type 'readonly Coord[]'. | TS2556 - A spread argument must either have a tuple type or be passed to a rest parameter.
+            return magnitude(vector(...coordsCopy));
+        },
+    );
+
+    _.each([0, 1], function (j) {
+        sides[j] = Math.round(sides[j]);
+    });
+
+    // Avoid degenerate triangles
+    if (
+        leq(sides[1] + sides[2], sides[0]) ||
+        leq(sides[0] + sides[2], sides[1]) ||
+        leq(sides[0] + sides[1], sides[2])
+    ) {
+        return startingPoint;
+    }
+
+    // Solve for angle by using the law of cosines
+    const innerAngle = lawOfCosines(sides[0], sides[2], sides[1]);
+
+    const outerAngle = GraphUtils.findAngle(
+        coordsCopy[rel(1)],
+        coordsCopy[rel(-1)],
+    );
+
+    const onLeft =
```

Comment:
Non-blocking nit: I'd love it if there could be a comment here explaining what the onLeft code is doing / how it affects the offset!